decorated_kanon.js

```diff
@@ -0,0 +1,172 @@
+/* "Decorated Kan-on" romanization
+ * - vowels based on romanized Kan-on
+ * - consonants based on romanized Go-on
+ * - distinctions not made in onyomi are marked with diacritics and digraphs
+ * More details at https://kanjisense.com/dict/middle-chinese-pronunciation
+ * @author justinsilvestre
+ */
+
+if (!音韻地位) return [
+  ['聲調', [1, '平:∅ 上:ˬ 去:ˎ 入:∅', '平:1 上:2 去:3 入:4']],
+];
+const toneDiacritics = 音韻地位.聲調 === '平:∅ 上:ˬ 去:ˎ 入:∅';
+const is = (...x) => 音韻地位.屬於(...x);
+
+let 聲母 = {
+  幫: 'p',   滂: 'pʻ',   並: 'b',   明: 'm',
+  端: 't',   透: 'tʻ',   定: 'd',   泥: 'n',  來: 'l',
+  知: 't',  徹: 'tʻ',  澄: 'd',  孃: 'n',
+  精: 'ts',  清: 'tsʻ',  從: 'dz',                     心: 's',  邪: 'z',
+  莊: 'tṣ', 初: 'tṣʻ', 崇: 'dẓ',                    生: 'ṣ', 俟: 'ẓ',
+  章: 'tś', 昌: 'tśʻ', 常: 'dź', 日: 'nź',          書: 'ś', 船: 'ź', 以: 'y',
+  見: 'k',   溪: 'kʻ',   羣: 'g',   疑: 'ng',
+  影: "ʾ",   曉: 'kh',    匣: 'gh',                                          云: '',
+}[音韻地位.母];
+
+let get韻母 = {
+  // 一等韻
+  東: () => {
+    if (is`入聲 三等`) {
+      if (is`幫組`) return 'ūk';
+      if (is`以母`) return 'ẁīk';
+      return is`莊組 或 章組` ? 'yūk' : 'wīk';
+    }
+    if (is`三等`) {
+      return is`幫組` ? 'ūng' : 'iūng';
+    }
```

Comment:
```suggestion
    if (is`三等`) {
      if (is`舒聲`) return is`幫組` ? 'ūng' : 'iūng';
      if (is`幫組`) return 'ūk';
      if (is`以母`) return 'ẁīk';
      return is`莊章組` ? 'yūk' : 'wīk';
    }
```
It seems like this is the only case that there are differences between 舒聲 and 入聲 (not including nasal ↔ plosive codas)?

Reply:
Yes, it's one of a few inconsistencies in attested Kan-on that I've incorporated in here, on a somewhat arbitrary basis. I had to make a judgment call as to which ones were worth keeping. I think the rationale here was, there are a fair amount of common characters with common Kan-on that reflected these differences.